src/box/replication.cc

```diff
@@ -971,6 +1009,8 @@ replicaset_connect(struct applier **appliers, int count,
 			break;
 		if (count - state.failed < replicaset_connect_quorum(count))
 			break;
+		if (state.ready)
+			break;
```

Comment:
Do you really need this flag? It seems you could simply walk here all the appliers and check if any of them is a connected bootstrap config leader. Wouldn't take much more code, but would allow to avoid adding more flags which need to be remembered to be updated.

Reply:
It seems I don't, thanks! I was having trouble with this code especially with `bootstrap_strategy = 'supervised'`. I was trying to go the other way and put everything to this flag: https://github.com/sergepetrenko/tarantool/commit/c3b471f7e454049a377cd7d69e3d82dd9f153539. It didn't turn out too well, and your approach looks promising. Will try.

Reply:
I reworked this part, but can't share a diff, it's rather dirty. Please check out the new approach in the commit itself.
There's also one extra commit: `428b74932 replication: refactor replicaset_connect()`